yarn-project/simulator/src/avm/avm_message_call_result.ts

```diff
@@ -1,29 +1,18 @@
 import { type Fr } from '@aztec/foundation/fields';
 
+import { type AvmRevertReason } from './errors.js';
+
 /**
  * Results of an contract call's execution in the AVM.
  */
 export class AvmContractCallResults {
-  public readonly reverted: boolean;
-  public readonly output: Fr[];
-
-  /** For exceptional halts */
-  public readonly revertReason: Error | undefined;
-
-  constructor(reverted: boolean, output: Fr[], revertReason?: Error) {
-    this.reverted = reverted;
-    this.output = output;
-    this.revertReason = revertReason;
-  }
+  constructor(public reverted: boolean, public output: Fr[], public revertReason?: AvmRevertReason) {}
 
-  /**
-   * Generate a string representation of call results.
-   */
   toString(): string {
-    let resultsStr = `reverted: ${this.reverted}, output: ${this.output}`;
+    let resultsString = `reverted: ${this.reverted}, output: ${this.output}`;
     if (this.revertReason) {
-      resultsStr += `, revertReason: ${this.revertReason}`;
+      resultsString += `, revertReason: ${this.revertReason!.message}`;
```

Comment:
I'm surprised you needed the ! given you have the conditional!

Reply:
TBH this toString changeset isn't really desired. I'm going to revert it back to what it is in master!